src/patch_parse.c

```diff
@@ -808,7 +808,7 @@ static int parse_patch_binary_side(
 
 		encoded_len = ((decoded_len / 4) + !!(decoded_len % 4)) * 5;
 
-		if (encoded_len > ctx->parse_ctx.line_len - 1) {
```

Comment:
I'd extract that as `static int parse_binary_patch_length(git_patch_parse_ctx *ctx, size_t decoded_len)`, just to make sure the overflow is handled only once, every time 😉. Not sure how generic that is, but maybe that can be made public at the parser layer (this idiom comes from libssh2, which looks at binary data *all the time*) ?

Reply:
Right now there's only a single location where we calculate the binary header length, and putting this into a more generic layer feels a bit premature to me. Binary patches use a special prefix-length-encoding that is currently in use for binary patches only, and I don't expect this to be used anywhere else anytime soon.

So I think we should definitely pull it out into a separate function as soon as there is a second user. But right now I don't quite think it to be worthwhile. Are you good with this?

Reply:
I'm fine with the current implementation, sorry for being overly pedantic 😉.

Reply:
No need to be sorry at all, I'm always happy to get your input!